backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/DockerCassandraRule.java

```diff
@@ -0,0 +1,139 @@
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.backends.cassandra;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.DockerClientFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.output.OutputFrame;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+import org.testcontainers.shaded.com.google.common.collect.ImmutableMap;
+
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.command.CreateVolumeCmd;
+import com.github.dockerjava.api.command.RemoveVolumeCmd;
+import com.github.dockerjava.api.model.Bind;
+import com.github.dockerjava.api.model.Binds;
+import com.github.dockerjava.api.model.Volume;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+
+public class DockerCassandraRule implements TestRule {
+
+    private static final Logger logger = LoggerFactory.getLogger(DockerCassandraRule.class);
+
+    private static final int CASSANDRA_PORT = 9042;
+    public static final String CASSANDRA_YAML = "$CASSANDRA_CONFIG/cassandra.yaml";
+
+    private final GenericContainer<?> cassandraContainer;
+    private final DockerClient client;
+    private final CreateVolumeCmd createTmpsFsCmd;
+    private final RemoveVolumeCmd deleteTmpsFsCmd;
+
+    private AtomicInteger nbTestBeforeRestart = new AtomicInteger(20);
+
+    public DockerCassandraRule() {
+        String tmpFsName = "cassandraData" + RandomStringUtils.randomAlphabetic(10);
+        client = DockerClientFactory.instance().client();
+        createTmpsFsCmd = client.createVolumeCmd()
+            .withName(tmpFsName)
+            .withDriver("local")
+            .withDriverOpts(
+                ImmutableMap.of(
+                    "type", "tmpfs",
+                    "device", "tmpfs",
+                    "o", "size=100m"));
+        deleteTmpsFsCmd = client.removeVolumeCmd(tmpFsName);
+        boolean deleteOnExit = false;
+        cassandraContainer = new GenericContainer<>(
+            new ImageFromDockerfile("cassandra_3_11", deleteOnExit)
+                .withDockerfileFromBuilder(builder ->
+                    builder
+                        .from("cassandra:3.11.0")
```

Comment:
Would be great to extract the Cassandra version

Reply:
not sure how it helps